pkg/ingest/store/factory.go

```diff
@@ -133,24 +133,24 @@ func (f *walkerFactory) buildGCSWalker(ctx context.Context) (*gcsWalker, error)
 }
 
 func (f *walkerFactory) buildAzureWalker() (*azureBlobWalker, error) {
-	var p pipeline.Pipeline
+	var c *service.Client
 	if f.params != nil {
 		azureParams, err := f.params.BlockstoreAzureParams()
 		if err != nil {
 			return nil, err
 		}
-		p, _, err = factory.BuildAzureClient(azureParams)
+		c, _, err = factory.BuildAzureServiceClient(azureParams)
 		if err != nil {
 			return nil, err
 		}
 	} else {
 		var err error
-		p, err = getAzureClient()
+		c, err = getAzureClient()
 		if err != nil {
 			return nil, err
 		}
 	}
-	return NewAzureBlobWalker(p)
+	return NewAzureBlobWalker(*c)
```

Comment:
I write the comment here but it is relevant for factory build also - we keep a copy of the client and not keeping the new client pointer.
Every time we create a client we copy the structure and not the pointer to the client we created.
In general if the function that creates the client can return an error which means that we keep a pointer (optional nil if we got an error) - I would keep that pointer and not copies.